docs/_howto/lolex-async-promises.md

```diff
@@ -0,0 +1,128 @@
+    const spy = sinon.spy();
+    maker.callAfterOneSecond(spy);
+
+    // callback is not called immediately
+    assert.ok(!spy.called);
+
+    // but it is called synchronously after the clock is fast forwarded
+    clock.tick(1000);
+    assert.ok(spy.called); // PASS
+});
+```
+
+We could expect similar behavior from a promisified timeout:
+
+```js
+// maker.js
+const util = require('util');
+const setTimeoutPromise = util.promisify(setTimeout);
+
+module.exports.asyncReturnAfterOneSecond = async () => {
+    await setTimeoutPromise(1000);
+    return 42;
+};
+```
+
+Trying a naive approach:
+
+```js
+// test.js
+it('should return 42 after one second', () => {
+    // result will hold a Promise after the next call
+    const result = maker.asyncReturnAfterOneSecond();
+
+    clock.tick(1000);
+
+    // the Promise was not resolved,
+    // even though we moved the time forward
+    assert.equal(result, 42); // FAIL
+});
+```
+
+The above test fails, since `asyncReturnAfterOneSecond` is an `async` function
+that returns a Promise, and it is currently impossible to control the Promises'
+forced resolution. **Their `then()` function always runs asynchronously**.
+
+It doesn't mean that `async` functions and Promises cannot be tested though.
+The most intuitive way to test the above `asyncReturnAfterOneSecond` is to
+simply wait for one second:
+
+```js
+// test.js
+
+// using async await
+it('should return 42 after one second', async () => {
+    const result = await maker.asyncReturnAfterOneSecond();
+    assert.equal(result, 42); // PASS
+});
+
+// or using Mocha's promises
+it('should return 42 after one second', () => {
+    const promise = maker.asyncReturnAfterOneSecond();
+
+    // this call does not really speed up anything
+    clock.tick(1000);
+
+    return promise.then(result => assert.equal(result, 42)); // PASS
+});
+```
+
+Although `async` functions cannot be tested synchronously, we can test Promises
+that are resolved in `setTimeout`. Consider the following function, that has the
+same functionality as `asyncReturnAfterOneSecond`:
+
+```js
+// maker.js
+module.exports.fulfillAfterOneSecond = () => {
+    return new Promise(fulfill => {
+        setTimeout(() => fulfill(42), 1000);
+    });
+};
+```
+
+`fulfillAfterOneSecond` resolves to 42 after one second, just like
+`asyncReturnAfterOneSecond`, but it is testable in a synchronous way:
+
+```js
+// test.js
+it('should be fulfilled after one second', () => {
+    const promise = maker.fulfillAfterOneSecond();
+
+    // this call actually makes the resolution quicker
+    clock.tick(1000);
+    return promise.then(result => assert.equal(result, 42)); // PASS
+});
+```
+
+The above test passes immediately, without the 1 second delay. This is because
+we do not try to hijack `then()` call, but use Mocha's native ability to test
+Promises, while speeding up their resolution.
+
+Although it is currently impossible to speed up `async` functions, or even
```

Comment:
This isn't correct. You can speed up `async` functions. It's just syntactic sugar-coating for promises. Anything you can do with promises is possible with async and vice versa. Unless you can inform me otherwise?

Reply:
It's easier to speed up promises because you can override the promise library to one that uses a scheduler

 Speeding up async functions without invoking the native `%RunMicrotasks` (which requires an addon) or using a `require('binding')` with an undocumented internal API requires transpiling your async functions to coroutines.

Reply:
True, but I didn't mean overriding the scheduler for the Promises, but rather the statement that implied that one cannot speedup `async` tests that use timers. It seemed as if the statement said that if you have an `async` test using `setTimeout` (for instance), then it would not be affected by `clock.tick` calls, which would be incorrect.

Just the fact that we are arguing the meaning might be a hint that this sentence should be rephrased somehow :-)

Reply:
Agreed